tests/scanners/test.base.js

```diff
@@ -13,6 +13,16 @@ class BaseScannerWithContents extends BaseScanner {
 
 describe('Base Scanner Class', function() {
 
+  it('scannerName should throw an error', () => {
+    try {
+      var scanner = new BaseScanner('', 'filename.txt');
+      scanner.scannerName;
```

Comment:
This seems wrong; accessing a property shouldn't throw like this. It seems very much like an unintended side-effect. I think it would be better if during initialisation of this class it checked/validated `this.scannerName` and threw during class creation so:

```js
var scanner = new BaseScanner('', 'filename.txt');
```
would `throw` without having to access the property.

Also you can use `const` here :-)